.github/workflows/canary-integration-test.yml

```diff
@@ -1698,3 +1698,52 @@ jobs:
         uses: ./.github/workflows/collect-logs
         with:
           name: ${{ github.job }}-${{ matrix.ceph-image }}
+
+  rgw-with-cephblockpool:
+    runs-on: ubuntu-22.04
+    if: "!contains(github.event.pull_request.labels.*.name, 'skip-ci')"
+    strategy:
+      matrix:
+        ceph-image: ${{ fromJson(inputs.ceph_images) }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+        with:
+          fetch-depth: 0
+
+      - name: consider debugging
+        uses: ./.github/workflows/tmate_debug
+        with:
+          use-tmate: ${{ secrets.USE_TMATE }}
+
+      - name: setup cluster resources
+        uses: ./.github/workflows/canary-test-config
+
+      - name: set Ceph version in CephCluster manifest
+        run: tests/scripts/github-action-helper.sh replace_ceph_image  "deploy/examples/cluster-test.yaml" "${{ github.event.inputs.ceph-image }}"
+
+      - name: validate-yaml
+        run: tests/scripts/github-action-helper.sh validate_yaml
+
+      - name: use local disk and create partitions for osds
+        run: |
+          tests/scripts/github-action-helper.sh use_local_disk
+          tests/scripts/github-action-helper.sh create_partitions_for_osds
+
+      - name: deploy cluster
+        run: tests/scripts/github-action-helper.sh deploy_cluster
+
+      - name: create CephBlockPool(s) and CephObjectStore
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+        run: |
+          kubectl create -f deploy/examples/rgw-with-cephblockpool.yaml
+
+      - name: wait for CephObjectStore to be ready
+        run: |
+          tests/scripts/validate_cluster.sh rgw rwc
```

Comment:
The RGW will create new pools that it needs but that Rook doesn't create itself. I'm thinking it could be helpful to check the pool list and make sure there aren't any unknown pools hanging around here. WDYT?

Reply:
I think the positive check is the important one I have in mind. Ceph just added 2 pools in Reef (v19), and an automated way to catch similar changes in the future would be great.

Reply:
As discussed on slack, probing ceph's pools revealed  unnecessary CRs being installed for all canary tests.  I've attempting to lay down the foundational to start cleaning  this up without unraveling the entire ball of string.